### DISCUSSION TITLE
Problems using Copilot with neovim / tmux

### DISCUSSION THREAD

**Discussion by phrazzld** [community/community#36558] [Copilot Conversations] [answered]
Neovim: 0.8.0
tmux: 3.3a

Installed Copilot with Packer. Now every time I launch neovim, I get the following error message:

```
Error detected while processing function <lambda>4[1]..<SNR>35_Start[4]..copilot#agent#New:
line   13:
E688: More targets than List items
```

This error occurs whenever I try to run a Copilot command as well, for instance `Copilot setup`.

Googling this error didn't yield many results. This six-year-old `vim-gitgutter` issue was the closest thing I could find, but I was unable to apply the fix here:
https://github.com/airblade/vim-gitgutter/issues/323

Has anyone else gotten Copilot to work with neovim and tmux? Has anyone else run into this issue and gotten past it?

**Comment by ahayworth** [comment, marked as answer]
@phrazzld You're seeing this bug because when you load vim and the plugin initializes, it attempts to detect which `node.js` version is available. If the plugin _cannot_ call `node --version` correctly, you are supposed to see an error like `Node.js returned error code <foo>` when you call `:Copilot setup`.

However, the error-handling code internally _itself_ has a bug; which is why you see this particular error. I can't open a PR upstream because it's limited to collaborators, but if @tpope wants to pull in [this fix](https://github.com/ahayworth/copilot.vim/commit/de4f4ea7a3ea43d706a49ec0f7a54e7a3aa54190) it will stop the plugin from crashing.

But you can fix your specific case by making sure you have a globally-accessible `node.js` installation available. I'm using `nodenv` personally, and forgot to set the global version - perhaps it's something similar for you.

**Reply by tpope** [reply]
Correct diagnosis, I'll see that it's fixed in the next release.

**Reply by phrazzld** [reply]
Thank you both!

**Comment by ossanna16** [comment]
Hey there @phrazzld and thank you for asking a good question! I'm glad you found our community 🙂
